public/systems_graphs.js

```diff
@@ -0,0 +1,209 @@
+
+        // Clear previous graphs before redrawing
+        d3.selectAll("svg").remove();
+
+        // Set the dimensions and margins of the graphs
+        const Margin = {top: 40, right: 100, bottom: 105, left: 70},
+            Width = 960 - Margin.left - Margin.right,
+            Height = 500 - Margin.top - Margin.bottom;
+
+        plotDiskMetrics(data);
+
+        function plotDiskMetrics(data) {
+            let diskMetrics = ["used", "free"]
+            // Create keys to stack
+            let diskKeys = [],
+            diskStr = ""
+
+            for (let i = 0; i < diskMetrics.length; i++) {
+                diskStr =  "disk_" + diskMetrics[i];
+                diskKeys.push(diskStr);
+            }
+
+            let colors = ["#228B22", "#fbdb9c"]
+            // color palette
+            let color = d3.scaleOrdinal().domain(diskMetrics).range(colors);
+
+            let stackDisk = d3.stack().keys(diskKeys),
+                diskDataStacked = stackDisk(data);
+
+            let svg = d3.select(".chart")
+                .append("svg")
+                .attr("width", Width + Margin.left + Margin.right + 120)
+                .attr("height", Height + Margin.top + Margin.bottom)
+                .append("g")
+                .attr("transform", "translate(" + Margin.left + "," + Margin.top + ")");
+
+            // Add X axis
+            let x = d3.scaleTime().domain(d3.extent(data, d => new Date(d.datetime))).range([0, Width]);
+            let xAxis = svg.append("g")
+                 .attr("transform", "translate(0," + Height + ")")
+                 .call(d3.axisBottom(x).tickFormat(dayTimeFormat))
+            
+            // Rotate X axis ticks
+            xAxis.selectAll("text")
+                .style("text-anchor", "end")
+                .attr("dx", "-.8em")
+                .attr("dy", ".15em")
+                .attr("transform", "rotate(-65)");
+
+            // Add X axis label
+            svg.append("text")
+                .attr("text-anchor", "end")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Date (dd HH:mm)");
+
+            // Add Y axis label
+            svg.append("text")
+                .attr("transform", "rotate(-90)")
+                .attr("y", 0 - Margin.left)
+                .attr("x", 0 - Height / 2)
+                .attr("dy", "1em")
+                .style("text-anchor", "middle")
+                .text("GB")
+                
+            let yLimit = data[0].disk_total
+            // Add Y axis
+            let y = d3.scaleLinear()
+                .domain([0, yLimit])
+                .range([ Height, 0 ]);
+                svg.append("g")
+                .call(d3.axisLeft(y).ticks(5))
+
+            // Add a clipPath: everything out of this area won't be drawn.
```

Comment:
Suggestion: it is probably better to only create an MVP with no new features from what we currently have in other graphs till you finish refactoring the code. clipPath here is great but I think it can wait for that. I will leave it to you to decide.

Reply:
Tooltips would be great to have them now though.

Reply:
I'm working on the tooltip, it may take a little bit longer.

Reply:
If we remove clip path, we should remove also remove zooming effect because the work together.

Reply:
I was referring to both here. but happy for you to keep it if it does not make your work harder when refactoring.

Reply:
I will handle those features in the refactoring, The zooming feature helps to look at the data at specified times of the day.